src/main/scala/io/iohk/ethereum/consensus/ConsensusAdapter.scala

```diff
@@ -35,15 +50,54 @@ class ConsensusAdapter(
         if (isBlockADuplicate(block.header, bestBlock.header.number)) {
           log.debug("Ignoring duplicated block: {}", block.idTag)
           Task.now(DuplicateBlock)
+        } else if (blockchain.getChainWeightByHash(bestBlock.header.hash).isEmpty) {
+          // This part is not really needed except for compatibility as a missing chain weight
+          // would indicate an inconsistent database
+          returnNoTotalDifficulty(bestBlock)
         } else {
           doBlockPreValidation(block).flatMap {
             case Left(error) =>
               Task.now(BlockImportFailed(error.reason.toString))
             case Right(BlockExecutionSuccess) =>
-              consensus.evaluateBranch(Seq(block))
+              enqueueAndGetBranch(block, bestBlock.number) match {
+                case None =>
+                  Task.now(BlockEnqueued)
```

Comment:
Can you please add a comment explaining this scenario is when the block is enqueue but in a branch not rooted?   I had to go check why a None would end up in a `BlockEnqueued` :)